Objects/unionobject.c

```diff
@@ -504,6 +512,57 @@ static PyNumberMethods union_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+union_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
```

Comment:
It's been a while since I wrote the docs for PEP 604 so I've forgotten the exact reason, but union types cannot be instantiated using the ``types.Union`` constructor. https://docs.python.org/3.10/library/stdtypes.html#union-type

But we need to support this for pickling, so I need to dig up the history of why it's like that.

Reply:
I will try to find a way how we can pickle `types.Union` without defining `__new__`.

Reply:
You can maybe also try looking at the original reference implementation in the PEP and the comment chain on why it's like that. If it's not a strong reason, we can change the docs and implement using ``__new__`` and save your effort.

Reply:
It is a great idea, thanks)

Reply:
Oh no! I didn't mean to lead you down the `from_args` path :(. Sorry.

I read through most of the conversation at PEP 604's PR. It seems there's no `union_new` because [there was no need for it](https://github.com/python/cpython/pull/21515#issuecomment-675747681). I also re-read PEP 604 and it's not dictated by the PEP. This means it's an implementation detail.

In that case, could you please revert to using ``__new__``? We can follow up with a docs PR later. Sorry for any double-work caused on your part.